### PR TITLE
bug/PLAT-41206 : Ensure javascript is rendered when adding promotions with markup

### PR DESCRIPTION
**Pull request by shah-tejas**
[PLAT-41206](https://jira.attivio.com/browse/PLAT-41206) : Promotions with markup don't render correctly in SUIT

If the markup in promotions contain javascript code in `<script>`, the code is not rendered in the searchui.
To overcome this, the `<div>` wrapper for the markup in `<PlacementResult>` is changed to `<iframe>`. The markup is added in the body of the iframe to ensure script as well as html is rendered.

Additionally, iframe also required unique title which is created using the new `markupCount` prop passed to PlacementResult.

### CODE REVIEW

File: src/components/PlacementResult.js [hunk 1]
```diff
@@ -12,13 +12,16 @@ type PlacementResultProps = {
   imageUrl: string | null;
   /** The raw html markup to display. Optional. */
   markup: string | null;
+  /** The position of the markup element in the placements array */
```
Comment (luigivaldez) [inline src/components/PlacementResult.js]: I think we can get rid of the markupCount property... It's really only needed when creating the key to use for a possible array of `PlacementResult` components. Including it in the `<iframe>`'s title doesn't really mean anything to the user...

File: src/components/PlacementResult.js [hunk 2]
```diff
@@ -30,16 +33,41 @@ export default class PlacementResult extends React.Component<PlacementResultDefa
...
+        <body>
+          ${this.props.markup || ''}
+        </body>
+      </html>
+    `;
+    const iframeSource = getBlobURL(source);
+
+    return (
+      <iframe
+        title={`Markup Iframe ${this.props.markupCount}`}
```
Comment (luigivaldez) [inline src/components/PlacementResult.js]: We can probably change this to something a little more user friendly like "Promotion"... There's no reason it needs to be unique in the rare case there are multiple markup tags placements...

**Comment by shah-tejas** [comment]
Addressed review comments

File: src/components/PlacementResults.js [hunk 3]
```diff
@@ -22,8 +22,12 @@ export default class PlacementResults extends React.Component<void, {}, void> {
     if (response && response.placements && response.documents.length > 0) {
       const placements = response.placements;
       const results = [];
+      let markupCount = 0;
       placements.forEach((placement: Placement) => {
-        const key = `${placement.linkUrl ? placement.linkUrl : 'nourl'}-${placement.linkText ? placement.linkText : 'notext'}-${placement.imageUrl ? placement.imageUrl : 'noimg'}`; // eslint-disable-line max-len
+        if (placement.markup) {
+          markupCount += 1;
+        }
+        const key = `${placement.linkUrl ? placement.linkUrl : 'nourl'}-${placement.linkText ? placement.linkText : 'notext'}-${placement.imageUrl ? placement.imageUrl : 'noimg'}-${placement.markup ? `markup${markupCount}` : 'nomarkup'}`; // eslint-disable-line max-len
```
Comment (luigivaldez) [inline src/components/PlacementResults.js]: This line is (well, was before, too) very long, with all the ternary operators and nested string substitutions, it's not super clear what we're doing.
What would you think about breaking It up into pieces and then concatenating the pieces to get the final key?
E.g.:
```
const urlKey = placement.linkUrl ? placement.linkUrl : 'noUrl';
const textKey = placement.linkText ? placement.linkText : 'noText';
const imageKey = placement.imageUrl ? placement.imageUrl : 'noImage';
const markupKey = placement.markup ? `markup${markupCount}` :  'noMarkup';
const key = `${urlKey}-${textKey}-${imageKey}-${markupKey}`;
```

Comment (shah-tejas) [reply src/components/PlacementResults.js]: Yes makes sense. I just realized now that es-lint is disabled on this line. That is why it didn't warn me about the long line!!
Will make this change.

**Comment by luigivaldez** [comment]
Can you update the branch with the changes in master to validate that it is still happy? (I don't there are any conflicting changes, so...)

Also, can you add some comments on the ticket about the types of markup you've tried it with?

**Comment by shah-tejas** [comment]
Sure, I rebased this branch with master and will add the markups in the ticket.

**Comment by klk1010** [comment]
@shah-tejas, something is off about how the changes from master were brought in to this branch. Those changes should not be showing up in your diff. To keep the git history clean, I'm going to create a new branch based off this one with your changes before you merged and create a new PR. You don't need to do anything, just an FYI.

**Comment by klk1010** [comment]
Closing this PR in favor of https://github.com/attivio/suit/pull/144.

**Pull request closed without merging**